mqt/bench/utils/utils.py

```diff
@@ -116,10 +116,13 @@ def __init__(self, probability):
         self._theta_p = 2 * np.arcsin(np.sqrt(probability))
         self.ry(2 * self._theta_p, 0)
 
-    def power(self, k):
+    def __eq__(self, other):
+        return isinstance(other, BernoulliQ) and self._theta_p == other._theta_p
+
+    def power(self, power: float, matrix_power: bool = True):
```

Comment:
The variable `matrix_power` is never used, isn't it? What is the reason for adding it?

Reply:
The reason here was that PyCharm (and I believe CodeQL) was complaining that the method had a different signature in the base class. In order to resolve this, I changed the signature to match the one in the base class.